web/src/modules/topic/components/CriteriaTable/CriteriaTable.tsx

```diff
@@ -84,8 +79,11 @@ export const CriteriaTable = ({ problemNodeId }: Props) => {
   const criteria = nodeChildren.filter((node) => node.type === "criterion");
   const solutions = nodeChildren.filter((node) => node.type === "solution");
 
-  const rowData = buildRows(criteria, solutions, criterionSolutionEdges);
-  const columns = buildColumns(solutions);
+  const rowNodes = useSolutionsForColumns ? solutions : criteria;
+  const columnNodes = useSolutionsForColumns ? criteria : solutions;
```

Comment:
this is backwards (and default for `useSolutionsForColumns` should be true)
```suggestion
  const rowNodes = useSolutionsForColumns ? criteria : solutions;
  const columnNodes = useSolutionsForColumns ? solutions : criteria;
```